README.md

```diff
@@ -29,35 +29,110 @@ It must be configured before defining routes and other plugins in order to cover
   - `onResponse`
   - `onError`
 - Instruments automatically custom 404 Not Found handler
+- TODO: Clarify server shutdown/cleanup around otel metrics/traces
 
-Example:
+## Example
 
 ```js
-// ... in your OTEL setup
-const FastifyOtelInstrumentation = require('@fastify/otel');
+// otel.js
+// Sets up metrics, tracing, HTTP & Node runtime instrumentation, and host metrics.
+// Uses ConsoleSpanExporter for local debugging of spans.
+const { NodeSDK } = require('@opentelemetry/sdk-node')
+const { ConsoleSpanExporter } = require('@opentelemetry/sdk-trace-node')
+const { SimpleSpanProcessor } = require('@opentelemetry/sdk-trace-base')
+const { PrometheusExporter } = require('@opentelemetry/exporter-prometheus')
+const { HttpInstrumentation } = require('@opentelemetry/instrumentation-http')
+const { RuntimeNodeInstrumentation } = require('@opentelemetry/instrumentation-runtime-node')
+const { HostMetrics } = require('@opentelemetry/host-metrics')
+const { FastifyOtelInstrumentation } = require('@fastify/otel')
+const { metrics, trace } = require('@opentelemetry/api')
+const { resourceFromAttributes } = require('@opentelemetry/resources')
+const { SemanticResourceAttributes } = require('@opentelemetry/semantic-conventions')
+
+// Console exporter for development-time span inspection
+// We could use '@opentelemetry/exporter-trace-otlp-http' instead
+const traceExporter = new ConsoleSpanExporter()
+const spanProcessor = new SimpleSpanProcessor(traceExporter)
+
+const prometheusExporter = new PrometheusExporter({ port: 9091 })
+
+const sdk = new NodeSDK({
+  resource: resourceFromAttributes({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'my-service-name',
+  }),
+  spanProcessor,
+  metricReader: prometheusExporter,
+  instrumentations: [
+    new HttpInstrumentation(),
+    new RuntimeNodeInstrumentation(),
+  ],
+})
+
+await sdk.start()
 
-// If serverName is not provided, it will fallback to OTEL_SERVICE_NAME
-// as per https://opentelemetry.io/docs/languages/sdk-configuration/general/.
-const fastifyOtelInstrumentation = new FastifyOtelInstrumentation({ servername: '<yourCustomApplicationName>' });
-fastifyOtelInstrumentation.setTracerProvider(provider)
+const fastifyOtelInstrumentation = new FastifyOtelInstrumentation({
```

Comment:
Although is not necessary, is often a good practice to place the instrumentation within the instrumentations property when instantiating the Node trace, I'd suggest to follow that line

Reply:
#Looks like when you pass FastifyOtelInstrumentation in the instruments array, you don't need to call setTracerProvider manually. We should nudge in this direction, since calling this seems like a narrow use case.